src/Process/Runner/MutationTestingRunner.php

```diff
@@ -74,26 +74,34 @@ public function __construct(
     /**
      * @param Mutation[] $mutations
      */
-    public function run(array $mutations, int $threadCount, string $testFrameworkExtraOptions): void
+    public function run(iterable $mutations, int $threadCount, string $testFrameworkExtraOptions): void
     {
-        $this->eventDispatcher->dispatch(new MutantsCreationWasStarted(count($mutations)));
+        $this->eventDispatcher->dispatch(new MutantsCreationWasStarted(0));
+        $this->eventDispatcher->dispatch(new MutationTestingWasStarted(0));
 
-        $processes = array_map(
-            function (Mutation $mutation) use ($testFrameworkExtraOptions): MutantProcess {
-                $mutant = $this->mutantFactory->create($mutation);
+        $processes = take($mutations);
+        $processes->map(function (Mutation $mutation) use ($testFrameworkExtraOptions): MutantProcess {
+            $mutant = $this->mutantFactory->create($mutation);
 
-                $process = $this->processBuilder->createProcessForMutant($mutant, $testFrameworkExtraOptions);
+            $process = $this->processBuilder->createProcessForMutant($mutant, $testFrameworkExtraOptions);
 
-                $this->eventDispatcher->dispatch(new MutantWasCreated());
+            $this->eventDispatcher->dispatch(new MutantWasCreated());
 
-                return $process;
-            },
-            $mutations
-        );
+            return $process;
+        });
 
         $this->eventDispatcher->dispatch(new MutantsCreationWasFinished());
 
-        $this->eventDispatcher->dispatch(new MutationTestingWasStarted(count($processes)));
+        // We filter these here because it is beyond responsibilities of a process manager to care about types of processes
```

Comment:
Adding something like this above this line:
```
$processes = take($processes->toArray());
```
We can make the entire process as "sequential" as before. E.g. it'll buffer until it collects all mutations.